### DISCUSSION TITLE
How to improve the rate at which the OAP Server consumes Kafka?

### DISCUSSION THREAD

**Discussion by linshenfanying** [apache/skywalking#12764] [Q&A]
![pic3](https://github.com/user-attachments/assets/856d350e-f5dd-4b55-9fd0-563450ad2bd5)

**Comment by linshenfanying** [comment]
We have utilized SkyWalking for application tracing, and the corresponding architecture is shown in the diagram. A total of 220 services and approximately 1000 instances have been integrated. We have noticed that the consumption rate of Kafka by SkyWalking is very slow, with a persistent backlog, and the consumption capacity often drops suddenly. We have also implemented cluster separation.

1. The configuration of our OAP Server is 2 CPU cores and 4 GB of memory. We would like to ask for optimization suggestions.

2. How can we improve the rate at which SkyWalking OAP Server consumes Kafka?
What are the recommended resource configurations for the OAP Server? Is the OAP Server’s performance more related to CPU or memory?

**Reply by wu-sheng** [reply]
OAP is very CPU and memory required applications. You should grant all resources.

**Reply by wu-sheng** [reply]
We exposed Kafka client configurations, if you prefer Kafka, you could setup accordingly.

https://github.com/apache/skywalking/blob/5c5274b93a5a12ebab62edea4cca7246366e992a/oap-server/server-starter/src/main/resources/application.yml#L385-L396
